worlds/sa2b/Regions.py

```diff
@@ -511,35 +596,101 @@ def connect_regions(world, player, gates: typing.List[LevelGate], cannon_core_em
         connect(world, player, names, LocationName.gate_0_region, shuffleable_regions[gates[0].gate_levels[i]])
 
     if len(gates) >= 2:
-        connect(world, player, names, 'Menu', LocationName.gate_1_region,
+        connect(world, player, names, 'Menu', LocationName.gate_1_boss_region,
                 lambda state: (state.has(ItemName.emblem, player, gates[1].gate_emblem_count)))
+
+        if gate_bosses[1] == all_gate_bosses_table[king_boom_boo]:
```

Comment:
Not a fan of the duplicated if for the king boom boo boss. Might be better to figure out which connector has that boss and then set the rule for that specifically.

Reply:
I don't love this either. The issue is that King Boom Boo won't always be present, and it needs to make sure that any levels that get placed past him are blocked by the rule. I tried a couple of other solutions, but didn't really come up with anything I liked any better.